src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -133,8 +135,8 @@ public function terminate(Request $request, Response $response)
             return;
         }
 
-        if ($this->getHttpKernel() instanceof TerminableInterface) {
-            $this->getHttpKernel()->terminate($request, $response);
+        if ($this->httpKernel instanceof TerminableInterface) {
```

Comment:
the first change which could be done very easily would be to create a local variable with the HttpKernel here, to avoid calling it twice.